lib/ret_web/controllers/api/v1/avatar_controller.ex

```diff
@@ -42,10 +42,19 @@ defmodule RetWeb.Api.V1.AvatarController do
   defp create_or_update(conn, params, avatar, account) do
     files_to_promote =
       (params["files"] || %{})
-      |> Enum.map(fn {k, v} -> {String.to_atom(k), List.to_tuple(v)} end)
+      |> Enum.map(fn
+        {k, nil} -> {String.to_atom(k), :remove}
+        {k, v} -> {String.to_atom(k), List.to_tuple(v)}
+      end)
       |> Enum.into(%{})
 
-    owned_file_results = Storage.promote(files_to_promote, account)
+    owned_file_results =
+      files_to_promote
+      |> Enum.map(fn
+        {k, {id, key, promotion_token}} -> {k, Storage.promote(id, key, promotion_token, account)}
+        {k, :remove} -> {k, {:ok, :remove}}
```

Comment:
Don't particularly love this but I want to be able to carry this map through without having to split off the removals